public/app/plugins/datasource/graphite/components/TagEditor.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useCallback, useMemo } from 'react';
+import { Dispatch } from 'redux';
+import { Segment, SegmentAsync } from '@grafana/ui';
+import { actions } from '../state/actions';
+import { GraphiteTag, GraphiteTagOperator } from '../types';
+import { getTagOperatorsSelectables, getTagsSelectables, getTagValuesSelectables } from '../state/providers';
+import { GraphiteQueryEditorState } from '../state/store';
+import { debounce } from 'lodash';
+
+type Props = {
+  tag: GraphiteTag;
+  tagIndex: number;
+  dispatch: Dispatch;
+  state: GraphiteQueryEditorState;
+};
+
+/**
+ * Editor for a tag at given index. Allows changing the name of the tag, operator or value. Tag names are provided with
+ * getTagsSelectables and contain only valid tags (it may depend on currently used tags). The dropdown for tag names is
+ * also used for removing tag (with a special "--remove tag--" option provided by getTagsSelectables).
+ *
+ * Options for tag names and values are reloaded while user is typing with backend taking care of auto-complete
+ * (auto-complete cannot be implemented in front-end because backend returns only limited number of entries)
+ */
+export function TagEditor({ dispatch, tag, tagIndex, state }: Props) {
+  const getTagsOptions = useCallback(
+    async (inputValue: string | undefined) => {
+      return await getTagsSelectables(state, tagIndex, inputValue || '');
+    },
```

Comment:
is async/await necessary here?
```suggestion
    (inputValue: string | undefined) => {
      return getTagsSelectables(state, tagIndex, inputValue || '');
    },
```

Reply:
- [ ] remove redundant async/await